include/tvm/runtime/packed_func.h

```diff
@@ -537,6 +538,42 @@ struct ObjectTypeChecker<Map<K, V>> {
   }
 };
 
+class TVMPODValue_;
+
+/*!
+ * \brief Type trait to specify special value conversion rules from
+ *     ObjectRef to primitive types.
+ *
+ *  TVM containers, such as tvm::runtime::Array, require the contained
+ *  objects to inherit from ObjectRef.  As a result, the wrapper types
```

Comment:
Trailing whitespace, also in lines 550, 552.

Reply:
Hmm, I'm not seeing the trailing whitespace in the commit, either using `git show --word-diff-regex="[ ]+|[^ ]+"`, setting my editor to show whitespace, or in the github diff.

Reply:
Sorry, I used the wrong definition, there is a double space before "As"

Reply:
Ah, I see.  This is intentional, but is more of a stylistic difference than anything else.  Traditionally, (accidental diversion into typography) a space after a sentence would use an em space (the same with as a capital M) after a sentence, wider than the spacing between words.  This isn't possible in monospaced fonts, so typewriters would emulate the em space by using two spaces.  Exactly which is better has since become the topic of a great many flame wars.

I was curious, and it looks like double-spacing after a sentence is twice as common as single-spacing in the TVM repo, with the Apache copyright header being the most prominent example.

```bash
# Count the number of periods sentences followed by one space
$ find . \
  \( -path ./3rdparty -o -path "./build*" \) -prune -o \
  \( -name "*.cc" -o -name "*.h" \) \
  -exec grep '[A-Za-z]\{2\}\. [A-Za-z]' {} /dev/null \; \
    | wc --lines
3958

# Count the number of periods sentences followed by two spaces
$ find . \
  \( -path ./3rdparty -o -path "./build*" \) -prune -o \
  \( -name "*.cc" -o -name "*.h" \) \
  -exec grep '[A-Za-z]\{2\}\.  [A-Za-z]' {} /dev/null \; \
    | wc --lines
7666
```